### PR TITLE
4.x : Add create action to model

### PR DESCRIPTION
**Pull request by deepakrkris**
Mixin ModelActions class to Model

### CODE REVIEW

File: lib/model-handler.js [hunk 1]
```diff
@@ -48,3 +49,20 @@ ModelHandler.findAllModels = function(workspace, cb) {
...
+
+class ModelActions {
+  create(cb) {
+    const workspace = this.getWorkspace();
+    const self = this;
+    const tasks = [];
+    tasks.push(function(next) {
+      fsUtility.writeModel(self, next);
+    });
+    workspace.execute(tasks, function(err) {
```
Comment (superkhau) [inline lib/model-handler.js]: `workspace.execute(tasks, cb);`

File: lib/util/mixin.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = function mixin(target, source) {
+  const attributes = Object.getOwnPropertyNames(source);
+  attributes.forEach(function(ix) {
```
Comment (superkhau) [inline lib/util/mixin.js]: use descriptive name instead of `ix` (`property`)

File: lib/util/mixin.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = function mixin(target, source) {
+  const attributes = Object.getOwnPropertyNames(source);
```
Comment (superkhau) [inline lib/util/mixin.js]: `properties` is probably a better name

File: lib/util/mixin.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  attributes.forEach(function(ix) {
+    if (ix === 'constructor') return;
```
Comment (superkhau) [inline lib/util/mixin.js]: typeof property?

File: lib/util/mixin.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    if (typeof source[ix] === 'function') {
```
Comment (superkhau) [inline lib/util/mixin.js]: ix -> property

File: lib/util/mixin.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+      const mx = source[ix];
```
Comment (superkhau) [inline lib/util/mixin.js]: rename mx to something, i don't even know what mx is -- maybe `value`?

**Comment by deepakrkris** [comment]
@slnode test please

File: lib/model-handler.js (hunk 1 above)
Comment (deepakrkris) [inline lib/model-handler.js]: agreed

**Comment by deepakrkris** [comment]
@superkhau fixed comments

**Comment by deepakrkris** [comment]
@slnode test please

File: test/unit/datamodel/model.js [hunk 7]
```diff
@@ -6,21 +6,42 @@
...
       const model = new Model(workspace, 'test', {}, {});
       expect(workspace.getNode('ModelDefinition', 'test')).to.eql(model);
     });
   });
 
-  function createWorkspace() {
-    workspace = new Workspace('/');
-    workspace.addDomain('ModelDefinition');
+  describe('create()', function() {
+    it('write model definition file in workspace', function(next) {
```
Comment (superkhau) [inline test/unit/datamodel/model.js]: creates the model definition file in the workspace
next -> done

File: test/unit/datamodel/model.js [hunk 8]
```diff
@@ -6,21 +6,42 @@
...
+      workspace = WorkspaceManager.getWorkspace();
+      const data = {name: 'test', facetName: 'common'};
+      const model =
+        new Model(workspace, 'common.models.test', data);
+      model.create(function(err) {
```
Comment (superkhau) [inline test/unit/datamodel/model.js]: if (err) return done(err);

File: test/unit/datamodel/model.js [hunk 9]
```diff
@@ -6,21 +6,42 @@
...
+        const file = model.getFilePath();
+        fs.readJson(file, function(err, data) {
+          if (err) return next(err);
```
Comment (superkhau) [inline test/unit/datamodel/model.js]: done(err);

File: test/unit/datamodel/model.js [hunk 10]
```diff
@@ -6,21 +6,42 @@
...
+          expect(Object.keys(data)).to.include.members(['facetName', 'name']);
+          next();
```
Comment (superkhau) [inline test/unit/datamodel/model.js]: done

**Pull request merged**